discord/automod.py

```diff
@@ -0,0 +1,420 @@
+        self.trigger: AutoModTrigger = AutoModTrigger.from_data(data['trigger_type'], data=data.get('trigger_metadata'))
+        self.enabled: bool = data['enabled']
+        self.raw_exempt_roles: List[int] = [int(role_id) for role_id in data['exempt_roles']]
+        self.raw_exempt_channels: List[int] = [int(channel_id) for channel_id in data['exempt_channels']]
+        self._actions: List[AutoModerationActionPayload] = data['actions']
+
+    def __repr__(self) -> str:
+        return f'<AutoModRule id={self.id} name={self.name!r} guild={self.guild!r}>'
+
+    def to_dict(self) -> AutoModerationRulePayload:
+        ret: AutoModerationRulePayload = {
+            'id': str(self.id),
+            'guild_id': str(self.guild.id),
+            'name': self.name,
+            'creator_id': str(self.creator_id),
+            'event_type': self.event_type.value,
+            'trigger_type': self.trigger.type.value,
+            'trigger_metadata': self.trigger.to_metadata_dict(),
+            'actions': [action.to_dict() for action in self.actions],
+            'enabled': self.enabled,
+            'exempt_roles': [str(role_id) for role_id in self.raw_exempt_roles],
+            'exempt_channels': [str(channel_id) for channel_id in self.raw_exempt_channels],
+        }  # type: ignore # trigger types break the flow here.
+
+        return ret
+
+    @property
+    def creator(self) -> Optional[User]:
+        """Optional[:class:`User`]: The user that created this rule."""
+        return self._state.get_user(self.creator_id)
+
+    @property
+    def exempt_roles(self) -> List[Role]:
+        """List[:class:`Role`]: The roles that are exempt from this rule."""
+        it = [role for role in (self.guild.get_role(role_id) for role_id in self.raw_exempt_roles) if role is not None]
+        return utils._unique(it)
+
+    @property
+    def exempt_channels(self) -> List[Union[GuildChannel, Thread]]:
+        """List[Union[:class:`abc.GuildChannel`, :class:`Thread`]]: The channels that are exempt from this rule."""
+        it = filter(None, map(self.guild._resolve_channel, self.raw_exempt_channels))
+        return utils._unique(it)
+
+    @property
+    def actions(self) -> List[AutoModRuleAction]:
+        """List[:class:`AutoModRuleAction`]: The actions that are taken when this rule is triggered."""
+        return [AutoModRuleAction.from_data(action) for action in self._actions]
+
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = None,
+        event_type: Optional[AutoModRuleEventType] = None,
+        actions: Optional[List[AutoModRuleAction]] = None,
+        enabled: Optional[bool] = None,
+        exempt_roles: Optional[List[Role]] = None,
+        exempt_channels: Optional[List[Union[GuildChannel, Thread]]] = None,
+    ) -> Self:
+        """|coro|
+
+        This method will edit an existing auto moderation rule.
```

Comment:
```suggestion
        Edit an existing auto moderation rule.
```

What permissions are needed to edit the rule? This should be documented.